### ISSUE TITLE
Sub-orchestration's FunctionFailedException.InnerException is null

### ISSUE THREAD

**Issue by kijujjav** [Azure/azure-functions-durable-extension#643]
**Issue**
I am throwing TaskCanceledException() from the action to the orchestrator and expecting FunctionFailedException.InnerException to have the exception, instead it has null.

**Expected**
FunctionFailedException.InnerException should have TaskCanceledException().

**Additional Details**

In action
throw new TaskCanceledException();

In Orchestrator
catch (Exception ex)
            {
                
                if (ex.InnerException is TaskCanceledException)
                {
                    log.LogError(ex.Message);
                }
             }
![InnerExceprionNull](https://user-images.githubusercontent.com/48494090/54231736-48f8f780-44c6-11e9-9ccb-ba525f8a7202.png)

**Comment by cgillum** [comment]
Do you get the same behavior with any other exception type? I wonder if `TaskCanceledException` is being inadvertently handled somewhere in the runtime.

**Comment by kijujjav** [comment]
InnerException is null no matter what type of exception I throw.

**Comment by cgillum** [comment]
I'm not able to reproduce this behavior:
![image](https://user-images.githubusercontent.com/2704139/54324040-23004f80-45b9-11e9-9805-b896a6745acb.png)

Can you share more details about your app? For example, the version of the Functions runtime (1.0 or 2.0) and the version of the nuget packages you are using?

**Comment by kijujjav** [comment]
We are calling context.CallSubOrchestratorAsync and the exception is thrown in the sub-orchestration, does the behavior differ for activity and sub-orchestration?

**Comment by kijujjav** [comment]
We are using Functions 2.0 and azure-functions-durable-extension 1.7.1

**Comment by cgillum** [comment]
I created the following test case orchestrator function to see whether this is reproduceable in the latest code:

```csharp
public static async Task SubOrchestrationThrow([OrchestrationTrigger] DurableOrchestrationContext ctx)
{
    string message = ctx.GetInput<string>();

    try
    {
        await ctx.CallSubOrchestratorAsync(nameof(TestOrchestrations.ThrowOrchestrator), message);
    }
    catch (FunctionFailedException e)
    {
        if (e.InnerException == null ||
            e.GetBaseException().GetType() != typeof(InvalidOperationException) ||
            !e.InnerException.Message.Contains(message))
        {
            throw new Exception("InnerException was not the expected value.");
        }

        // rethrow the original exception
        throw;
    }
}
```

The code worked as expected and the inner-exception was not null. I think for now we can assume that if there was an issue in v1.7.1 that it is fixed in v1.8.0.

**Comment by cgillum** [comment]
Resolved with [v1.8.0](https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.DurableTask/1.8.0) release.

**Comment by kijujjav** [comment]
Thanks Chris, I don't see the bug in 1.8 version
